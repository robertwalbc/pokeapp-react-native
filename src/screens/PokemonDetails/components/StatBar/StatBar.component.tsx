import React from 'react';
import { View } from 'react-native';
import { styles } from './StatBar.styles';

interface StatBarProps {
  value: number;
  color: string;
}

export const StatBar: React.FC<StatBarProps> = ({ value, color }) => (
  <View style={styles.statBar}>
    <View
      style={[
        styles.statFill,
        {
          width: `${Math.min((value / 255) * 100, 100)}%`,
          backgroundColor: color,
        },
      ]}
    />
  </View>
); 