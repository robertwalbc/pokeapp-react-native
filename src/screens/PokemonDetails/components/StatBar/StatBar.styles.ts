import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  statBar: {
    flex: 1,
    height: 10,
    backgroundColor: '#E0E0E0',
    borderRadius: 5,
    overflow: 'hidden',
    marginHorizontal: 8,
  },
  statFill: {
    height: '100%',
    borderRadius: 5,
  },
}); 