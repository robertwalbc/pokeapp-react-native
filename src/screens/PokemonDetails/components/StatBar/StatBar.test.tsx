import React from 'react';
import renderer from 'react-test-renderer';
import { StatBar } from './StatBar.component';

describe('StatBar', () => {
  it('renders the bar with correct width and color', () => {
    const tree = renderer.create(<StatBar value={100} color="#4CAF50" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
}); 