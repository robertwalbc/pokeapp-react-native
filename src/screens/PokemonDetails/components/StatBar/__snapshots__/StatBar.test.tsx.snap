// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatBar renders the bar with correct width and color 1`] = `null`;
